kythe/go/serving/pipeline/beam.go

```diff
@@ -313,12 +331,55 @@ func groupCrossRefs(key *spb.VName, refStream func(**ppb.Reference) bool, emitSe
 		g.Anchor = append(g.Anchor, ref.Anchor)
 	}
 
+	callers := make(map[string]*xspb.CrossReferences_Caller)
+	callsites := make(map[string][]*xspb.CrossReferences_Callsite)
+	var call *xspb.CrossReferences
+	for callStream(&call) {
+		switch e := call.Entry.(type) {
+		case *xspb.CrossReferences_Caller_:
+			callers[kytheuri.ToString(e.Caller.Caller)] = e.Caller
+		case *xspb.CrossReferences_Callsite_:
```

Comment:
Is the trailing underscore here and above intentional?

Reply:
Yeah, that's how proto generates oneof fields with fields named the same as a nested message.